### PR TITLE
Add download of correct target platform binary

### PR DESCRIPTION
**Pull request by lewismarshall**
Tested specifying "other" platform flags...

@vinn946 Be good to get this version checked running from osx- it's been created as https://github.com/appvia/artefactor/releases/tag/v0.1.1-rc5

### CODE REVIEW

File: pkg/cmd/artefactor.go [hunk 1]
```diff
@@ -23,8 +23,13 @@ const (
 	FlagWebFiles = "web-files"
 	// FlagLogs enabled debug logs
 	FlagLogs = "logs"
+	// FlagTargetPlatform allows the correct version of artefactor to be saved
+	// with files.
+	FlagTargetPlatform = "target-platform"
 	// DefaultArchiveDir
 	DefaultArchiveDir = "downloads"
+	// The defualt binary type to include in downloads
```
Comment (gambol99) [inline pkg/cmd/artefactor.go]: s/defualt/default/ .. and for a sake of a linter add `DefaultTargetPlatform` start of comment

File: pkg/cmd/save.go [hunk 2]
```diff
@@ -101,15 +110,86 @@ func save(c *cobra.Command) error {
...
-			saveDir,
-			me,
-			err)
+	// Save the binary for the target platform
+	platform := c.Flag(FlagTargetPlatform).Value.String()
+	return saveMe(saveDir, platform)
+}
+
+// saveMe saves a copy of the target binary in the save dir
+func saveMe(saveDir string, platform string) error {
```
Comment (gambol99) [inline pkg/cmd/save.go]: `func saveMe(saveDir, platform string) error`

File: pkg/web/save.go [hunk 3]
```diff
@@ -37,6 +37,30 @@ func Save(
...
+
+	// Now the file is updated - update the checksum...
+	hash, err := hashcache.UpdateCache(download)
+	if err != nil {
+		return err
+	}
+	if !hashcache.IsCachedMatch(download, sha256) {
+		log.Printf("invalid checksum (%s) for %s, expecting %q", sha256, download, hash)
+		return errors.Errorf("invalid checksum for %s", download)
+	} else {
```
Comment (gambol99) [inline pkg/web/save.go]: we could drop the else .. as the return negates it ..

File: pkg/cmd/artefactor.go (hunk 1 above)
Comment (lewismarshall) [reply pkg/cmd/artefactor.go]: Added checks in build and fixed this

File: pkg/cmd/save.go (hunk 2 above)
Comment (lewismarshall) [reply pkg/cmd/save.go]: Fixed this. (and added linters)

File: pkg/web/save.go (hunk 3 above)
Comment (lewismarshall) [reply pkg/web/save.go]: totes - nice spot.

File: pkg/docker/names.go [hunk 4]
```diff
@@ -31,7 +31,7 @@ func FilePathToImageName(fileName string) (imageName string, err error) {
 		dirL := len(imageDirName)
```
Comment (gambol99) [inline pkg/docker/names.go]: does this section of code just want the filename stripping away the dir? ... or i'm i reading wrong .. you can just use `filepath.Base(filename)` if so

Comment (lewismarshall) [reply pkg/docker/names.go]: @gambol99 - totally, must have done this with headphones off - updated.

**Pull request merged**